packages/opentelemetry-exporter-collector/src/CollectorExporter.ts

```diff
@@ -14,9 +14,10 @@
  * limitations under the License.
  */
 
+import { Attributes, Logger } from '@opentelemetry/api';
 import { ExportResult, NoopLogger } from '@opentelemetry/core';
 import { ReadableSpan, SpanExporter } from '@opentelemetry/tracing';
-import { Attributes, Logger } from '@opentelemetry/api';
+import * as grpc from 'grpc';
```

Comment:
CollectorExporter is used by web and node. grpc is not part of web so this won't be working fine and also credentials are not needed in web so far. 
I think ExporterCollector should be treated as base class and not be exported. Then inside folder platform inside node and browser you should create a new files ExporterCollector. The web can simply return the "base" ExporterCollector but the node you can then simply extend the base with all needed functionality that you added here. Then export the extended class correctly

Reply:
I understand the credentials will only work in node, but it won't break web right?

Reply:
it will break as `grpc` is a node package, cannot be included into web

Reply:
unless we just let credentials to exists in both, but then explain that you should not use it which might be misleading why. By splitting this it will be easier to add stuff I think

Reply:
It is type only and is compiled away. In recent versions of ts there is even a way to enforce this by doing `import type { ChannelCredentials } from 'grpc';`

Reply:
I'm affraid that once we allow having anything from grpc here it is like asking for more trouble in future - who will remember then that we should not do that. Splitting this now will save us troubles later and it will be easier to maintain it too.

Reply:
@obecny I've refactored it as you suggested. Please check if web/node exports are done correctly, I'm not sure I understand the initial idea with separate exports for each env. Let me know if you have any other comments.

Reply:
Also if you wanna do any changes yourself - feel free to push.